lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -115,7 +116,7 @@ def eval_block
 
         def verify_message
           return true if @expected_message.nil?
-          values_match?(@expected_message, @actual_error.message)
+          values_match?(@expected_message, @actual_error.message.to_s)
```

Comment:
I've added a spec, but FWIW it's `Composable` thats causing the problem which this surfaces, not setting the exception message to a string basically.